File: frontend/src/lib/api/onboarding.ts
/**
 * Onboarding API methods.
 */

import apiClient from "./client";

export interface OnboardingStep {
  step: string;
  name: string;
  completed: boolean;
  current: boolean;
}

export interface OnboardingStatus {
  completed: boolean;
  current_step: number;
  steps: OnboardingStep[];
}

export interface OnboardingCompleteResponse {
  message: string;
  redirect_url: string;
}

/**
 * Get current onboarding status.
 */
export async function getOnboardingStatus(): Promise<OnboardingStatus> {
  const response = await apiClient.get<OnboardingStatus>(
    "/api/v1/onboarding/status"
  );
  return response.data;
}

/**
 * Complete the onboarding process.
 */
export async function completeOnboarding(): Promise<OnboardingCompleteResponse> {
  const response = await apiClient.post<OnboardingCompleteResponse>(
    "/api/v1/onboarding/complete"
  );
  return response.data;
}

/**
 * Skip onboarding entirely.
 */
export async function skipOnboarding(): Promise<OnboardingCompleteResponse> {
  const response = await apiClient.post<OnboardingCompleteResponse>(
    "/api/v1/onboarding/skip"
  );
  return response.data;
}

File: src/schemas/onboarding.py
"""
Onboarding-related Pydantic schemas.
"""

from typing import List

from pydantic import Field

from src.schemas.common import BaseSchema


class OnboardingStep(BaseSchema):
    """Single onboarding step status."""

    step: str
    name: str
    completed: bool
    current: bool = False


class OnboardingStatus(BaseSchema):
    """Response schema for onboarding status endpoint."""

    completed: bool
    current_step: int = Field(..., ge=1, le=3)
    steps: List[OnboardingStep]


class OnboardingCompleteResponse(BaseSchema):
    """Response schema for completing onboarding."""

    message: str = "Onboarding completed successfully"
    redirect_url: str = "/dashboard"
